sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -25,53 +26,68 @@ enum {
  */
 typedef struct pinmux_input {
   top_earlgrey_pinmux_peripheral_in_t periph;
-  top_earlgrey_pinmux_insel_t pad_insel;
+  top_earlgrey_pinmux_insel_t insel;
   top_earlgrey_muxed_pads_t pad;
 } pinmux_input_t;
 
 /**
  * An MIO pad and a peripheral output to link it to.
  */
 typedef struct pinmux_output {
-  top_earlgrey_pinmux_mio_out_t pad;
-  top_earlgrey_pinmux_outsel_t periph;
+  top_earlgrey_pinmux_mio_out_t mio;
+  top_earlgrey_pinmux_outsel_t outsel;
+  top_earlgrey_muxed_pads_t pad;
 } pinmux_output_t;
 
 /**
  * UART RX pin.
  */
 static const pinmux_input_t kInputUart0 = {
     .periph = kTopEarlgreyPinmuxPeripheralInUart0Rx,
-    .pad_insel = kTopEarlgreyPinmuxInselIoc3,
+    .insel = kTopEarlgreyPinmuxInselIoc3,
     .pad = kTopEarlgreyMuxedPadsIoc3,
 };
 
 /**
  * UART TX pin.
  */
 static const pinmux_output_t kOutputUart0 = {
-    .pad = kTopEarlgreyPinmuxMioOutIoc4,
-    .periph = kTopEarlgreyPinmuxOutselUart0Tx,
+    .mio = kTopEarlgreyPinmuxMioOutIoc4,
+    .outsel = kTopEarlgreyPinmuxOutselUart0Tx,
+    .pad = kTopEarlgreyMuxedPadsIoc4,
 };
 
 /**
  * SW strap pins.
- *
  */
+#define PINMUX_ASSERT_EQ_(a, b) \
+  static_assert((a) == (b), "Unexpected software strap configuration.")
+
+PINMUX_ASSERT_EQ_(SW_STRAP_0_PERIPH, kTopEarlgreyPinmuxPeripheralInGpioGpio22);
+PINMUX_ASSERT_EQ_(SW_STRAP_0_INSEL, kTopEarlgreyPinmuxInselIoc0);
+PINMUX_ASSERT_EQ_(SW_STRAP_0_PAD, kTopEarlgreyMuxedPadsIoc0);
```

Comment:
No assertions for straps 1 and 2?

Reply:
We have, but they are right above the corresponding definitions, please see couple of line below.